pom.xml

```diff
@@ -781,31 +774,25 @@
     <profile>
       <!-- This profile is used to enable GraalVM native image testing -->
       <id>native</id>
-
       <dependencies>
-
         <dependency>
-          <groupId>org.junit.vintage</groupId>
-          <artifactId>junit-vintage-engine</artifactId>
-          <version>5.9.2</version>
-          <scope>test</scope>
-        </dependency>
-
-        <dependency>
-          <groupId>org.graalvm.buildtools</groupId>
-          <artifactId>junit-platform-native</artifactId>
-          <version>0.9.20</version>
-          <scope>test</scope>
+          <groupId>org.opentest4j</groupId>
+          <artifactId>opentest4j</artifactId>
+          <version>1.2.0</version>
         </dependency>
       </dependencies>
-
       <build>
         <plugins>
           <plugin>
             <groupId>org.apache.maven.plugins</groupId>
             <artifactId>maven-surefire-plugin</artifactId>
-            <!-- Must use older version of surefire plugin for native-image testing. -->
-            <version>2.22.2</version>
+            <dependencies>
+              <dependency>
```

Comment:
Move `junit-vintage-engine` inside `maven-surefire-plugin` so that the plugin would use `surefire-junit4` provider instead of the `surefire-junit-platform` provider.

[reference](https://maven.apache.org/surefire/maven-surefire-plugin/examples/junit-platform.html#junit4-api-in-test-dependencies)